src/module_2/groceries_analysis.md

```diff
@@ -0,0 +1,1811 @@
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+    </tr>
+    <tr>
+      <th>1958</th>
+      <td>33e0cb6eacea0775e34adbaa2c1dec16b9d6484e6b9324...</td>
+      <td>Top Up</td>
+      <td>UKD</td>
+      <td>2022-03-09 23:12:25</td>
+      <td>2022-03-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+    </tr>
+    <tr>
+      <th>675</th>
+      <td>57ca7591dc79825df0cecc4836a58e6062454555c86c35...</td>
+      <td>Top Up</td>
+      <td>UKI</td>
+      <td>2021-04-23 16:29:02</td>
+      <td>2021-04-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+    </tr>
+    <tr>
+      <th>4694</th>
+      <td>085d8e598139ce6fc9f75d9de97960fa9e1457b409ec00...</td>
+      <td>Top Up</td>
+      <td>UKJ</td>
+      <td>2021-11-02 13:50:06</td>
+      <td>2021-11-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+    <class 'pandas.core.frame.DataFrame'>
+    Index: 4983 entries, 2160 to 3360
+    Data columns (total 10 columns):
+     #   Column                 Non-Null Count  Dtype  
+    ---  ------                 --------------  -----  
+     0   user_id                4983 non-null   object 
+     1   user_segment           4983 non-null   object 
+     2   user_nuts1             4932 non-null   object 
+     3   first_ordered_at       4983 non-null   object 
+     4   customer_cohort_month  4983 non-null   object 
+     5   count_people           325 non-null    float64
+     6   count_adults           325 non-null    float64
+     7   count_children         325 non-null    float64
+     8   count_babies           325 non-null    float64
+     9   count_pets             325 non-null    float64
+    dtypes: float64(5), object(5)
+    memory usage: 428.2+ KB
+    
+    Mising values(NAs):
+
+
+
+    user_id                     0
+    user_segment                0
+    user_nuts1                 51
+    first_ordered_at            0
+    customer_cohort_month       0
+    count_people             4658
+    count_adults             4658
+    count_children           4658
+    count_babies             4658
+    count_pets               4658
+    dtype: int64
+
+
+## Check Orders 
+
+With the orders dataframe, we could try to identify the purchasing patterns of customers. Both, the number of orders per user and the number of items per order are important metrics to understand the purchasing patterns of customers. 
+
+
+
+Let's first check the frequency of orders per user.
+
+
+```python
+orders_frequency = orders_df['user_id'].value_counts().reset_index().rename(columns={'count': 'order_count'})
+display(orders_frequency['order_count'].describe())
+plt.hist(orders_frequency["order_count"],bins = len(orders_frequency["order_count"].unique()), edgecolor="k")
```

Comment:
As in PR1 I would recommend the use of fig, ax and use the axes

Reply:
You are not following the PEP8 style guide. For example, missing a space before "bins", the spaces arund "=" of bins should be removed

Reply:
About the PEP8 style, you're right. I forgot to run black formatter.